Update the getting and setting data guide. (#16533)
ckeditor/ckeditor5: docs/getting-started/setup/getting-and-setting-data.md

# Getting and setting data

> CKEditor&nbsp;5 allows you to retrieve the data from and save it to your server (or to your system in general) in various ways. In this guide, you can learn about the available options along with their pros and cons.

Reviewer: ```suggestion
In CKEditor&nbsp;5 there are several ways to load and save data. In this guide, you can learn about the available options and their advantages and disadvantages.
```

The original sentence suggested, that it is the CKEditor that handles retrieving and saving the data to and from the database/backend, but that's not always the case (e.g. if someone doesn't use our services).

## Autosaving

> One of the best ways to enhance the user experience in saving editor data is by providing an autosave functionality. This ensures users do not need to remember to save their data, and it prevents your system from being flooded with save events since the save action will be debounced.

Reviewer: ```suggestion
One of the best ways to improve the user experience is to automatically save editor data as it changes. This ensures that users do not have to remember to save their data and prevents work loss.
```

IMO we don't need the information about debouncing, as it is an implementation details and is already mentioned in the linked document.

We provide and autosave feature that automatically saves the data (for example sends it to the server) when needed. Please refer to the {@link features/autosave Autosave} guide for details.

## Initializing the editor with data

> By default, the editor is initialized with the content of the element on which it is initialized:

Reviewer: ```suggestion
By default, the editor has the content of the DOM element on which it was initialized on.
```

```html
<div id="editor">
	<!-- This content will appear in the editor if you initialize with this element. -->
	<p>Hello, world!</p>
</div>
```

> If you have the data in the JavaScript layer or cannot alter the HTML, the editor’s API provides the `initialData` configuration property. This allows you to provide the content during the initialization of the editor:

Reviewer: ```suggestion
However, if you cannot alter the HTML or you load the data asynchronously using JavaScript, you can use the `initialData` configuration property to set the initial state of the editor.
```

```js
ClassicEditor
	.create( document.querySelector( '#editor' ), {
		plugins: [ /* ... */ ],
		toolbar: [ /* ... */ ],
		initialData: '<p>Hello, world!</p>'
	} )
	.then( /* ... */ )
	.catch( /* ... */ );
```

The {@link module:core/editor/editorconfig~EditorConfig.initialData `initialData`} property will initialize the editor with the provided data, overriding the content provided at the HTML level.

If you are setting up the editor with integrations like {@link getting-started/integrations/react React}, consult the documentation for additional properties provided to initialize the data.

## Getting the editor data with `getData()`

> If the editor content needs to be retrieved on demand for any reason, such as for sending it to the server through an Ajax call, use the `getData()` method:

Reviewer: ```suggestion
You can use the `getData()` method, if you need to get the editor content on demand, such as for sending it to the server using JavaScript APIs.
```

CKEditor 5 allows you to retrieve the data from and save it to your server (or to your system in general) in various ways. In this guide, you can learn about the available options along with their pros and cons.

## Autosaving

One of the best ways to enhance the user experience in saving editor data is by providing an autosave functionality. This ensures users do not need to remember to save their data, and it prevents your system from being flooded with save events since the save action will be debounced.

The autosave feature we provide automatically saves the data (e.g., sends it to the server) when needed. Please refer to the {@link features/autosave Autosave} guide for details.

## Initializing the editor with data

> By default, the editor is initialized with the content of the element on which it is initialized:

Reviewer 2: Not sure what this would me, exactly.

Author: Fixed by Filip.
